### PR TITLE
Added announcement banner

### PR DESCRIPTION
**Pull request by JeffSchering**
This is for [PLT-3466 E10: Add announcement bar feature #6509](https://github.com/mattermost/platform/pull/6509)

### CODE REVIEW

File: source/administration/announcement-banner.rst [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+===================
+
+In Mattermost Enterprise Edition, an announcement banner is available that allows system administrators to display a notice that's visible to all users on the system.
+
+.. image:: ../images/announcement-banner-1106x272.png
+  :width: 1106
+  :height: 272
+  :alt: Shows the announcement banner at the top of the user's screen.
+
+By default, users can dismiss the banner until you either change the text of the banner or until you re-activate the banner. A setting in the System Console allows you to prevent users from dismissing the banner. You can also control the text color and the background color.
```
Comment (lindy65) [inline source/administration/announcement-banner.rst]: We're saying `users` but then talk about `you`...  Should `you` not be `they`?

Comment (JeffSchering) [inline source/administration/announcement-banner.rst]: The "you" here is the administrator.

File: source/administration/announcement-banner.rst [hunk 2]
```diff
@@ -0,0 +1,20 @@
+Announcement Banner
+===================
+
+In Mattermost Enterprise Edition, an announcement banner is available that allows system administrators to display a notice that's visible to all users on the system.
```
Comment (esethna) [inline source/administration/announcement-banner.rst]: Please see config settings doc for how we typically format things that are enterprise only

Comment (esethna) [inline source/administration/announcement-banner.rst]: system administrators > System Admins

File: source/administration/announcement-banner.rst (hunk 1 above)
Comment (esethna) [inline source/administration/announcement-banner.rst]: Agree with LI. I don't think we typically refer to you as admins. We should state the role explicitly I think

Comment (esethna) [inline source/administration/announcement-banner.rst]: Propose we link to config settings doc banner settings?

Comment (esethna) [inline source/administration/announcement-banner.rst]: "until you either change the text of the banner or until you re-activate the banner" > We should just change this to state that the admin can chose to make the banner permanent (can't be dismissed), if set in the console.

File: source/administration/announcement-banner.rst [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+.. image:: ../images/announcement-banner-1106x272.png
+  :width: 1106
+  :height: 272
+  :alt: Shows the announcement banner at the top of the user's screen.
+
+By default, users can dismiss the banner until you either change the text of the banner or until you re-activate the banner. A setting in the System Console allows you to prevent users from dismissing the banner. You can also control the text color and the background color.
+
+**To activate the banner**:
+
+1. In the GENERAL section of the System Console, click **Policy**.
```
Comment (esethna) [inline source/administration/announcement-banner.rst]: I think we typically refer to system console settings with the following formatting: **System Console** > **Policy**

File: source/administration/announcement-banner.rst [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+2. In the *Enable Announcement Banner* section, click **true**.
+3. In the *Banner Text* field, enter the text of the announcement that you want to make.
+4. Set the background and text colors.
+5. Optional. To prevent users from dismissing the banner, in the *Allow Banner Dismissal* section, click **false**.
```
Comment (esethna) [inline source/administration/announcement-banner.rst]: What does optional mean? If the banner is enabled then they must choose false or true

File: source/administration/config-settings.rst [hunk 5]
```diff
@@ -441,7 +441,57 @@ Set the time limit that users have to edit their messages after posting.
 | The second setting is ``"PostEditTimeLimit": 300`` with whole number input. To enable ``PostEditTimeLimit``, set ``AllowEditPost`` to ``time_limit``.         |
 +---------------------------------------------------------------------------------------------------------------------------------------------------------------+
 
-________
+Enable Announcement Banner
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Enable an announcement banner across all teams. The banner is displayed at the top of the screen and is the entire width of the screen. By default, users can dismiss the banner until you either change the text of the banner or until you re-activate the banner. You can prevent users from dismissing the banner, and you can control the text color and the background color.
```
Comment (esethna) [inline source/administration/config-settings.rst]: I'm not sure what "re-activate" means in this context

File: source/administration/config-settings.rst [hunk 6]
```diff
@@ -441,7 +441,57 @@ Set the time limit that users have to edit their messages after posting.
...
+The color of the text in the announcement banner.
+
++-------------------------------------------------------------------------------------------------+
+| This feature's ``config.json`` setting is ``""BannerTextColor": "#333333"`` with string input.  |
++-------------------------------------------------------------------------------------------------+
+
+Allow Banner Dismissal
+^^^^^^^^^^^^^^^^^^^^^^
+
+**True**: Users can dismiss the banner until its next update.
```
Comment (esethna) [inline source/administration/config-settings.rst]: What happens if users refresh their page or logout and log in again?

File: source/administration/announcement-banner.rst (hunk 1 above)
Comment (JeffSchering) [inline source/administration/announcement-banner.rst]: This doc is in the administration section, and it is read by administrators, so the second person is appropriate here, and in accordance with the style guide.

File: source/administration/announcement-banner.rst (hunk 2 above)
Comment (JeffSchering) [inline source/administration/announcement-banner.rst]: ok

File: source/administration/announcement-banner.rst (hunk 3 above)
Comment (JeffSchering) [inline source/administration/announcement-banner.rst]: It's less ambiguous to do it this way, and easier to navigate. When there are duplicate names, then it's not clear which one to use. For example **System Console > Email** makes you wonder if it's the **Email** in the AUTHENTICATION section or in the NOTIFICATIONS section.

File: source/administration/announcement-banner.rst (hunk 4 above)
Comment (JeffSchering) [inline source/administration/announcement-banner.rst]: Optional means that they can prevent users from dismissing the banner or not; it's their choice. The default setting is true.

File: source/administration/config-settings.rst (hunk 5 above)
Comment (JeffSchering) [inline source/administration/config-settings.rst]: ok

File: source/administration/config-settings.rst (hunk 6 above)
Comment (JeffSchering) [inline source/administration/config-settings.rst]: ok

File: source/administration/announcement-banner.rst [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+Available in `Enterprise Edition E10 and higher <https://about.mattermost.com/pricing/>`_.
+
+In Mattermost Enterprise Edition, an announcement banner is available that allows System Admins to display a notice that's visible to all users on the system.
+
+.. image:: ../images/announcement-banner-1106x272.png
+  :width: 1106
+  :height: 272
+  :alt: Shows the announcement banner at the top of the user's screen.
+
+By default, users can dismiss the banner until they log in again or until you update the banner. You can update the banner by either changing the text of the banner or by re-enabling the banner after it has been disabled. You can also control the text color and the background color. A setting in the System Console allows you to prevent users from dismissing the banner.
```
Comment (lfbrock) [inline source/administration/announcement-banner.rst]: Would it be possible to add something here that's more around use cases than just describing functionality? 

I think we were trying to move towards phrasing docs in a way that answer user questions as opposed to just listing features, and including information about why people would use the feature might help with that.

File: source/administration/announcement-banner.rst (hunk 4 above)
Comment (esethna) [inline source/administration/announcement-banner.rst]: But in that sense all of the settings are optional (ie they could leave it as the default color). Can we just reword this to remove the word "optional"?

File: source/administration/announcement-banner.rst (hunk 7 above)
Comment (JeffSchering) [inline source/administration/announcement-banner.rst]: Sure. What use cases do you have in mind?

File: source/administration/announcement-banner.rst (hunk 4 above)
Comment (JeffSchering) [inline source/administration/announcement-banner.rst]: ok

**Comment by esethna** [comment]
Approving this PR so it can get merged into the 3.10 branch. Propose we make the improvement LB mentioned against master after the 3.10 branch is merged
